pkg/server/options/options.go

```diff
@@ -311,7 +320,44 @@ func (o *Options) Complete(ctx context.Context, rootDir string) (*CompletedOptio
 		o.GenericControlPlane.ServiceAccountSigningKeyFile = o.Controllers.SAController.ServiceAccountKeyFile
 	}
 
-	completedGenericOptions, err := o.GenericControlPlane.Complete(ctx, nil, nil)
+	// o.GenericControlPlane.Complete creates self-signed certificates
+	// with the advertise address by default. This can cause spurious
+	// errors if the server binds on multiple interfaces but only
+	// advertises for one.
```

Comment:
```suggestion
	// o.GenericControlPlane.Complete creates self-signed certificates
	// with the advertise address by default. This can cause spurious
	// errors if the server binds on multiple interfaces.
```